### PR TITLE
Initial websocket communication

### PR DESCRIPTION
**Pull request by SietzeKoonstra**

### CODE REVIEW

File: CSS Server/Models/CameraConnection.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    break;
+                case WebSocketMessageType.Close:
+                    Console.WriteLine("close message received maar ik verwacht dat hij hier nooit komt aangezien ik hem al eerder controleer");
+                    break;
+            }
+        }
+
+        public bool IsOnline()
+        {
+            if(_webSocket != null && _webSocket.State == WebSocketState.Open)
```
Comment (JohannesKauffmann) [inline CSS Server/Models/CameraConnection.cs:54]: `return (_webSocket != null && _webSocket.State == WebSocketState.Open)` ?

File: CSS Server/Models/Camera.cs [hunk 2]
```diff
@@ -4,16 +4,25 @@ public class Camera
...
-        /// <summary>
-        /// Now a simple getter and setter, can be replaced in the future with logic checking the websocketConnection
-        /// </summary>
-        public bool Connected { get; set; }
-
+        public bool Connected
+        {
+            get 
+            {
+                if (CameraConnection != null && CameraConnection.IsOnline())
```
Comment (JohannesKauffmann) [inline CSS Server/Models/Camera.cs:18]: `return (CameraConnection != null && CameraConnection.IsOnline())` ?

File: CSS Server/Models/CameraManager.cs [hunk 3]
```diff
@@ -38,5 +41,57 @@ public List<Camera> Cameras
...
+
+            if (isIncorrectRequest)
+                return null;
+
+            //Convert JTokens to .NET types. 
+            int id = JId.ToObject<int>();
+            string password = jPassword.ToObject<string>();
+
+            //Get the camera with its id
+            Camera camera = FindCamera(id);
```
Comment (JohannesKauffmann) [inline CSS Server/Models/CameraManager.cs:78]: Maybe `this.cameras.FirstOrDefault` with Linq instead of loop?

Comment (TimGels) [reply CSS Server/Models/CameraManager.cs:82]: Can be resolved I think?

**Review by JohannesKauffmann — APPROVED** [review]
lgtm

File: CSS Server/Models/CameraManager.cs [hunk 4]
```diff
@@ -38,5 +45,52 @@ public List<Camera> Cameras
...
+        /// 
+        /// </summary>
+        /// <param name="webSocket"></param>
+        /// <returns></returns>
+        /// <exception cref="NotImplementedException"></exception>
+        public async Task<Camera> ValidateCameraConnection(WebSocket webSocket)
+        {
+            //Read the first message of the connection for validation purposes.
+            byte[] buffer = new byte[1024 * 4];
+            WebSocketReceiveResult firstResult = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
```
Comment (TimGels) [inline CSS Server/Models/CameraManager.cs:58]: The code scanner complains about useless warning.

**Pull request merged**